includes/admin/class-wc-rest-upe-flag-toggle-controller.php

```diff
@@ -0,0 +1,105 @@
+<?php
+/**
+ * Class WC_REST_UPE_Flag_Toggle_Controller
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * REST controller for UPE feature flag.
+ */
+class WC_REST_UPE_Flag_Toggle_Controller extends WP_REST_Controller {
+	const UPE_CHECKOUT_EXPERIENCE_FLAG = 'upe_checkout_experience_enabled';
+
+	/**
+	 * Endpoint namespace.
+	 *
+	 * @var string
+	 */
+	protected $namespace = 'wc/v3';
+
+	/**
+	 * Endpoint path.
+	 *
+	 * @var string
+	 */
+	protected $rest_base = 'wc_stripe/upe_flag_toggle';
+
+	/**
+	 * Verify access to request.
+	 */
+	public function check_permission() {
+		return current_user_can( 'manage_woocommerce' );
+	}
+
+	/**
+	 * Configure REST API routes.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			[
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => [ $this, 'get_flag' ],
+				'permission_callback' => [ $this, 'check_permission' ],
+			]
+		);
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			[
+				'methods'             => WP_REST_Server::EDITABLE,
+				'callback'            => [ $this, 'set_flag' ],
+				'permission_callback' => [ $this, 'check_permission' ],
+				'args'                => [
+					'is_upe_enabled' => [
+						'description'       => __( 'Determines if the UPE feature flag is enabled.', 'woocommerce-gateway-stripe' ),
+						'type'              => 'boolean',
+						'validate_callback' => 'rest_validate_request_arg',
+					],
+				],
+			]
+		);
+	}
+
+	/**
+	 * Retrieve flag status.
+	 *
+	 * @return WP_REST_Response
+	 */
+	public function get_flag() {
+		$settings = get_option( 'woocommerce_stripe_settings' );
+
+		return new WP_REST_Response(
+			[
+				'is_upe_enabled' => empty( $settings[ self::UPE_CHECKOUT_EXPERIENCE_FLAG ] ) ? 'no' : $settings[ self::UPE_CHECKOUT_EXPERIENCE_FLAG ],
+			]
+		);
+	}
+
+	/**
+	 * Update the data.
+	 *
+	 * @param WP_REST_Request $request Full data about the request.
+	 */
+	public function set_flag( WP_REST_Request $request ) {
+		$is_upe_enabled = $request->get_param( 'is_upe_enabled' );
+
+		if ( null === $is_upe_enabled ) {
+			return new WP_REST_Response( [ 'result' => 'bad_request' ], 400 );
+		}
+
+		$settings = get_option( 'woocommerce_stripe_settings' );
+
+		if ( $is_upe_enabled ) {
```

Comment:
It looks like the `get_flag` returns a string `'yes'`/`'no``. But this method accepts a boolean to determine if the same flag should be enabled/disabled.

What do you think about ensuring that both methods accept/return consistent values (either both accept booleans or both accept `'yes'`/`'no'` strings)?

Reply:
Updating API payload to `{"is_upe_enabled": "yes"}` instead of boolean.